tempodb/blocklist/poller.go

```diff
@@ -154,9 +223,19 @@ func (p *Poller) pollBlock(ctx context.Context, tenantID string, blockID uuid.UU
 	}
 
 	if err != nil {
-		metricBlocklistErrors.WithLabelValues(tenantID).Inc()
 		return nil, nil, err
 	}
 
 	return blockMeta, compactedBlockMeta, nil
 }
+
+func (p *Poller) buildTenantIndex() bool {
+	for i := 0; i < p.cfg.TenantIndexBuilders; i++ {
+		job := jobPrefix + strconv.Itoa(i)
+		if p.sharder.Owns(job) {
```

Comment:
This is more of a question on the intended behavior: it looks all index builders poll and write the index for all tenants continuously.  That seems ok from a redundancy standpoint (i.e. if one builder stops working, the system still has fresh data).  Sharding by tenant here seems beneficial for scaling up to many tenants, but an unhealthy compactor ring entry would also impact polling.

Reply:
As discussed offline this approach is simpler operationally and we will go with it for now. The mentioned per tenant sharding is preferable, however, in the long term. I will make an issue to add it after this PR is merged.